lib/wax/metadata.ex

```diff
@@ -327,6 +288,12 @@ defmodule Wax.Metadata do
     }
   end
 
+  @spec build_metadata_statement_entry(map()) :: Wax.Metadata.Statement.t()
+  defp build_metadata_statement_entry(%{"metadataStatement" => metadata_statement}) do
+    {:ok, statement} = Wax.Metadata.Statement.from_json(metadata_statement)
```

Comment:
For MDSv3, I'm inclined to stop parsing metadata into Elixir structures, because Metadata format can change over time (new fields are added) and it causes this process to crash when it happens.

I think JSON-decoding and keeping metadata for each entry as-is is sufficient.